core/src/main/java/org/elasticsearch/action/support/replication/ReplicationRequest.java

```diff
@@ -177,6 +177,11 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeString(index);
     }
 
+    @Override
+    public Task createTask(long id, String type, String action) {
+        return new ReplicationTask(id, type, action, this::getDescription, getParentTaskNode(), getParentTaskId());
```

Comment:
I'm a bit hesitant about introducing this by default to all indexing requests- I guess Rally will tell us whether this has implications. Note though that getDescription calles request::toString  which in the case of IndexRequest forces a rendering of the document source to JSON - something we don't want to dot:

```
   String sSource = "_na_";
        try {
            sSource = XContentHelper.convertToJson(source, false);
        } catch (Exception e) {
            // ignore
        }
```

ShardBulkRequest is more reasonable...